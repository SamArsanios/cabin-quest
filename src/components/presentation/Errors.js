import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Alert, Button } from 'react-bootstrap';
import { unloadError } from '../../redux/actions/utilityActions';

class Errors extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: true,
    };
  }

  componentWillUnmount() {
    const { unloadError } = this.props;
    unloadError();
  }

  render() {
    const { show } = this.state;
    const { errors } = this.props;

    const setShow = () => {
      const { unloadError } = this.props;
      unloadError();
      this.setState({
        show: false,
      });
    };

    return (
      <div>
        <Alert show={show} variant="danger">
          <Alert.Heading>Sorry Something went wrong!</Alert.Heading>
          <div>
            {errors.request && (
            <h6 className="my-4">{errors.request.response}</h6>
            )}

            {errors.message && (
            <h6 className="my-4">
              {errors.message}
              {' '}
            </h6>
            )}
            {errors.response && (
            <h6 className="my-4">{errors.response.data.error}</h6>
            )}

            {errors.response && errors.response.status === 401 && (
            <h6 className="my-4 text-center">
              Your Login session has expired. Kindly login again.
            </h6>
            )}

            {errors.response && (
            <h6 className="my-4">{errors.response.data.error}</h6>
            )}
            <h6 className="content">
              If you are trying to login double check your username,email and
              password
            </h6>
          </div>
          <hr />
          <div className="d-flex justify-content-end">
            <Button onClick={setShow} variant="outline-danger">
              close
            </Button>
          </div>
        </Alert>
      </div>
    );
  }
}

Errors.propTypes = {
  errors: PropTypes.shape,
  unloadError: PropTypes.func.isRequired,
};

Errors.defaultProps = {
  errors: PropTypes.shape,
};

const mapStateToProps = (state) => ({
  errors: state.error.err,
});

export default connect(mapStateToProps, { unloadError })(Errors);
