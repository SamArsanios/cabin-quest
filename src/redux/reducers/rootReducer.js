import { combineReducers } from 'redux';
import { errorReducer, successReducer } from './utilityReducer';
import userReducer from './userReducer';
import cabinReducer from './cabinReducer';
import favReducer from './favReducer';
import cabinDetailReducer from './cabinDetailReducer';

const rootReducer = combineReducers({
  userData: userReducer,
  data: cabinReducer,
  cabin: cabinDetailReducer,
  favourite: favReducer,
  succMsg: successReducer,
  error: errorReducer,
});

export default rootReducer;
