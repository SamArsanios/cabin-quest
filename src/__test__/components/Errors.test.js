import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import { composeWithDevTools } from 'redux-devtools-extension';
import Errors from '../../components/presentation/Errors';
import rootReducer from '../../redux/reducers/rootReducer';

const state = {};
const middleware = [thunk];
const store = createStore(rootReducer, state, composeWithDevTools(applyMiddleware(...middleware)));
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Errors', () => {
  it('renders correctly', () => {
    const tree = render(<Errors />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});