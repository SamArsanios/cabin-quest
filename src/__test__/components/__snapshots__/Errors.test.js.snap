// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="fade alert alert-danger show"
          role="alert"
        >
          <div
            class="alert-heading h4"
          >
            Sorry Something went wrong!
          </div>
          <div>
            <h6
              class="content"
            >
              If you are trying to login double check your username,email and password
            </h6>
          </div>
          <hr />
          <div
            class="d-flex justify-content-end"
          >
            <button
              class="btn btn-outline-danger"
              type="button"
            >
              close
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="fade alert alert-danger show"
        role="alert"
      >
        <div
          class="alert-heading h4"
        >
          Sorry Something went wrong!
        </div>
        <div>
          <h6
            class="content"
          >
            If you are trying to login double check your username,email and password
          </h6>
        </div>
        <hr />
        <div
          class="d-flex justify-content-end"
        >
          <button
            class="btn btn-outline-danger"
            type="button"
          >
            close
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
